### PR TITLE
CyberArk AIM v2

### PR DESCRIPTION
**Pull request by ohaim1008**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/27829
https://github.com/demisto/etc/issues/25758
https://github.com/demisto/etc/issues/23288

## Description
Rewrite of the CyberArk AIM integration in python, with the relevant commands only.
The other commands in the integration that uses the PAS API moved to the CyberArk PAS Integration (https://github.com/demisto/etc/issues/27830)
Deprecate the old integration

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ohaim1008** [comment]
Missing - readme and explanation about:
* The rewrite in python and the separation of the AIM and PAS commands to different integrations.
* The credentials that are currently used in order to perform NTLM autonthication if needed.

File: Packs/cyberark_AIM/Integrations/CyberArkAIMV2/CyberArkAIMV2.yml
Comment (glicht) [inline Packs/cyberark_AIM/Integrations/CyberArkAIMV2/CyberArkAIMV2.yml:58]: I think it will be better to use: `demisto/ntlm`. py3ews has a lot of libs you don't need. See: https://github.com/demisto/dockerfiles/blob/repository-info/demisto/ntlm/last.md

File: Packs/cyberark_AIM/Integrations/CyberArkAIMV2/CyberArkAIMV2_description.md
Comment (glicht) [inline Packs/cyberark_AIM/Integrations/CyberArkAIMV2/CyberArkAIMV2_description.md:1]: Need to update this with proper description.

**Review by glicht — CHANGES_REQUESTED** [review]

File: Packs/cyberark_AIM/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.yml
Comment (bakatzir) [inline Packs/cyberark_AIM/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.yml:31]: try to not have this key first, I think it might create problem in the unifier.

**Comment by ohaim1008** [comment]
The build is failing when the 2 integration, the current one written in java and the v2 are in the same pack, therefore, moved to a different pack.

**Comment by ohaim1008** [comment]
@glicht added the Certificate Based Authentication, please take another look.

**Review by glicht — CHANGES_REQUESTED** [review]
Looks good. 

Please add at the end of main a finally clause which deletes the temporary certificate files.

File: Packs/CyberArkAIMv2/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py
Comment (DeanArbel) [inline Packs/CyberArkAIMv2/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py:143]: `LOG` is being deprecated. It is suggested to use `demisto.debug` directly instead.

File: Packs/CyberArkAIMv2/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py
Comment (DeanArbel) [inline Packs/CyberArkAIMv2/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py:162]: This can also be `demisto.error`

**Review by DeanArbel — COMMENTED** [review]
Good work. Please go over my comments.

File: Packs/CyberArkAIMv2/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py
Comment (glicht) [inline Packs/CyberArkAIMv2/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py:157]: Close is not enough. You still need to delete the file. Also, check if cf is not None.

File: Packs/CyberArkAIMv2/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py
Comment (DeanArbel) [inline Packs/CyberArkAIMv2/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py:160]: I don't think `os.close` is needed, but `os.remove` should be used here.

File: Packs/CyberArkAIMv2/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py
Comment (DeanArbel) [inline Packs/CyberArkAIMv2/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py:163]: I don't think `os.close` is needed, but `os.remove` should be used here.

File: Packs/CyberArkAIMv2/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py
Comment (DeanArbel) [inline Packs/CyberArkAIMv2/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py:165]: I believe `return_error` is a better option here, as it'll run a `demisto.results` with an error entry object.

**Review by DeanArbel — COMMENTED** [review]
Almost done, just a few small notes.

**Review by glicht — CHANGES_REQUESTED** [review]
@DeanArbel @ohaim1008 @bakatzir  we want v2 integrations to be in the same pack as the v1 was in. Please add the v2 integration to the original AIM pack.

File: Packs/CofenseTriage/.pack-ignore
Comment (glicht) [inline Packs/CofenseTriage/.pack-ignore:1]: Why this modification?

**Comment by glicht** [comment]
@ohaim1008 you are renaming the pack directory. I think this can be problematic as the directory name is used for the ID of the pack. I suggest using the original directory name of the pack.

**Comment by ohaim1008** [comment]
@glicht I will, the integration wasn't in a folder in the integrations directory and it looks like every change there is an issue.
so I'll add the new integration to the integrations directory but leave the old one as it was before.

**Comment by ohaim1008** [comment]
@glicht we are having issue with the build - the integration files of the v1 integration are not in a dir,
if I put them in a dir the id is changing as you wrote before... wdut?

**Comment by ohaim1008** [comment]
The 2 integrations are in the same pack, waiting for final review,
build is green.

File: Packs/cyberark_AIM/ReleaseNotes/1_0_1.md
Comment (bakatzir) [inline Packs/cyberark_AIM/ReleaseNotes/1_0_1.md:6]: maybe we should split it to 

##### CyberArk AIM v2

##### CyberArk AIM

@kirbles19 WDYT?

**Review by bakatzir — APPROVED** [review]

File: Packs/cyberark_AIM/ReleaseNotes/1_0_1.md
Comment (glicht) [reply Packs/cyberark_AIM/ReleaseNotes/1_0_1.md:6]: We should only write for the v2 if there are no changes for the v1.

**Comment by bakatzir** [comment]
@glicht Any other comments? if not please consider approving.

File: Packs/cyberark_AIM/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py
Comment (glicht) [inline Packs/cyberark_AIM/Integrations/CyberArkAIM_v2/CyberArkAIM_v2.py:151]:
```suggestion
            raise NotImplementedError(f'{command} is not an existing CyberArk AIM command')
```

**Review by glicht — APPROVED** [review]

**Comment by kirbles19** [comment]
@ohaim1008 If this requires tech docs review again, please remove the `docs-approved` label.

**Comment by ohaim1008** [comment]
@kirbles19 sure, it would be great if @ShirleyDenkberg could take another look at the release notes (only the last line was changes) and in `integration-cyberark_AIM.yml` the description of the integration was changes - we are deprecating it.

**Comment by ShirleyDenkberg** [comment]
@ohaim1008 Review is done.

**Comment by clay584** [comment]
@ohaim1008 @glicht I just saw this show up in XSOAR and the previous integration was marked as deprecated. Is the intention to provide the same XSOAR commands in `CyberArk AIM v2` that were available in `CyberArk AIM (Deprecated)`?

With the v2 integration currently, only the command `cyberark-aim-list-credentials` is exposed, and it does not provide the actual secret, just a listing of accounts.

So, I am wondering if I should continue to use `CyberArk AIM (Deprecated)` until `CyberArk AIM v2` has feature parity?

Thank you.

**Comment by ohaim1008** [comment]
Hi @clay584,
The intension was to separate the CyberArk AIM and CyberArk PAS commands and to give customers more certification options (such as windows certification).
The `cyberark-aim-list-credentials` is the new version of `list-credentials` and does the same, we don't show the exact password in the context (it is highly insecure), you can use the `fetch credentials` option and use the credential.
The commands `account-details` and `reset-credentials` are available in the new integration `CyberArk PAS` along with a lot of other commands for this product.

**Pull request merged**
